### PR TITLE
Fix the issue with Prometheus scraper for managed cluster

### PR DESCRIPTION
**Pull request by bhatpmk**
# Description

This change has a fix to scrape Prometheus metrics from managed cluster, by utilizing the _**/federate**_ endpoint. The match[] argument is set to federate any metric name. The configuration **_honor_labels_** is set to true, to not overwrite any labels exposed by the source server.

Fixes VZ-2448

# Checklist 

As the author of this PR, I have:

- [x] Checked that I included or updated copyright and license notices in all files that I altered
- [ ] Added or updated unit tests for any new functions I added
- [ ] Added or updated integration tests if appropriate
- [x] Added or updated acceptance tests if appropriate

Code reviewer, please confirm this PR:

- [ ] Addressed the requirement and meets the acceptance criteria
- [ ] Does not introduce unrelated or spurious changes
- [ ] Does not introduce any unapproved dependency
- [ ] Makes sense and it easy to understand, and/or difficult areas of code are clearly documented so that they can be understood

### CODE REVIEW

**Review by michael-cico — DISMISSED** [review]

File: platform-operator/controllers/clusters/sync_prometheus.go [hunk 1]
```diff
@@ -28,6 +28,11 @@ const (
 scrape_interval: 20s
 scrape_timeout: 15s
 scheme: https
+honor_labels: true
```
Comment (pfmackin) [inline platform-operator/controllers/clusters/sync_prometheus.go:31]: How is conflict avoided when metrics on the master and admin clusters have the same name?

Comment (jmaron99) [reply platform-operator/controllers/clusters/sync_prometheus.go:31]: The scraped metrics have an additional label indicating the managed cluster from which they were scraped.

File: tests/e2e/pkg/common.go [hunk 2]
```diff
@@ -255,7 +255,7 @@ func MetricsExist(metricsName, key, value string) bool {
 
 // MetricsExist validates the availability of a given metric in the given cluster
 func MetricsExistInCluster(metricsName, key, value, kubeconfigPath string) bool {
-	metric, err := QueryMetric(metricsName, getPrometheusIngressHost(kubeconfigPath))
+	metric, err := QueryMetric(metricsName, getPrometheusIngressHost(kubeconfigPath), kubeconfigPath)
```
Comment (jmaron99) [inline tests/e2e/pkg/common.go:258]: just a minor comment, but it seems we could keep the arguments to two:  metricName and kubeConfig, and then leverage getPrometheusIngressHost within QueryMetric to obtain the host there

**Review by jmaron99 — APPROVED** [review]

File: tests/e2e/pkg/common.go (hunk 2 above)
Comment (bhatpmk) [reply tests/e2e/pkg/common.go]: @jmaron99 I have addressed the review comment.

**Pull request merged**